### PR TITLE
Hk246 bbox comb

### PR DESCRIPTION
**Pull request by vskorole**

### CODE REVIEW

**Comment by jgallagher59701** [comment]
We need to fix the tests before merging this branch. Here is the log: https://api.travis-ci.org/v3/job/447967172/log.txt

**Comment by vskorole** [comment]
Found 
637: Comparing dods response for dmrpp/s3_dmrpp_join_new.ncml to baseline ${abs_srcdir}/baselines/dmrpp/s3_dmrpp_join_new.ncml  FAILED (dmrpp_tests.at:13)

File: functions/BBoxCombFunction.cc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    unsigned int rnk1= 0;
+    unsigned int rnk2 = 0;
+
+    switch (argc) {
+    case 0:
+    case 1:
+        // Must have 2 arguments
+        throw Error(malformed_expr, wrong_args);
+
+    default:
```
Comment (jgallagher59701) [inline functions/BBoxCombFunction.cc]: Use 'case 2:' and not default and have default: throw an exception. As the code is now, the number of args can be 3, 4, ..., although only the first two will be used.

File: functions/BBoxCombFunction.cc [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+#include "BBoxCombFunction.h"
+#include "roi_util.h"
+
+using namespace std;
+using namespace libdap;
+
+namespace functions {
+
+/**
+ * @brief Combine two bounding boxes with different shapes, forming their combination.
```
Comment (jgallagher59701) [inline functions/BBoxCombFunction.cc]: shorten @brief to just "Combine two bounding boxes with different shapes"

File: functions/BBoxCombFunction.cc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ *
+ * This combines 2 BBox variables (Array of Structure) forming
```
Comment (jgallagher59701) [inline functions/BBoxCombFunction.cc]: Change to:
This combines to bounding boxes (Array of Structure) of rank N and M into a single bounding box of rank N+M.

File: functions/unit-tests/BBoxCombFunctionTest [hunk 4]
```diff
@@ -0,0 +1,210 @@
+#! /bin/sh
```
Comment (jgallagher59701) [inline functions/unit-tests/BBoxCombFunctionTest]: This is a generated file. It should not be in git

**Comment by jgallagher59701** [comment]
oh, this should be merged!

**Pull request merged**